internal/pkg/remote/client.go

```diff
@@ -247,6 +247,10 @@ func (c *Client) checkApiVersionHeaders(reqID string, resp *http.Response) {
 // URI returns the remote URI.
 func (c *Client) URI() string {
 	host := c.config.GetHosts()[0]
+	if strings.HasPrefix(host, string(ProtocolHTTPS)+"://") ||
+		strings.HasPrefix(host, string(ProtocolHTTP)+"://") {
+		return host + "/" + c.config.Path
+	}
```

Comment:
Nit: wouldn't at this point be better to use a [url.Parse](https://pkg.go.dev/net/url#Parse) to extract the various parts and then [url.JoinPath](https://pkg.go.dev/net/url#URL.JoinPath) the configured path ?

Reply:
I'm wondering why have you added this extra case. Is it a valid case the protocol on `host` to be optional?

Normally I'd agree with Paolo, but as this function does not return an error and [url.Parse](https://pkg.go.dev/net/url#Parse) returns one, it'd not be possible to handle the error here. Therefore, I like how you did it.